### DISCUSSION TITLE
Reconnecting to / reattaching an ssh domain

### DISCUSSION THREAD

**Discussion by mkuratczyk** [wezterm/wezterm#3538] [Q&A] [answered]
I work on two machines and have two workspaces: local and a remote ssh domain. I have a keymap to switch between them and it works perfectly (better workflow than I had before switching to `wezterm`, so thanks!). However, the remote domain gets detached sometimes, when my laptop is closed, wifi changes, etc. In such cases, `SwitchWorkspaceRelative` does nothing, because there is no other workspace. Is it possible to either:
1. Have a reconnect mechanism so that `wezterm` just makes sure that my remote machine is always available
2. Or have a keymap that will switch to the other workspace and attach the domain if it's not currently attached (a keymap to a custom lua function?)

Right now I just have another keymap for attaching the remote domain so it's still pretty good, but I'd love it to be attached automatically when I want to use it.

Thanks,

**Comment by wez** [comment]
Are you using full on multiplexing with the ssh domain (with wezterm-mux-server installed and running on the remote host), or are you using it in the ad-hoc ssh connection mode?

**Reply by mkuratczyk** [reply]
I'm using wezterm multiplexing with the remote domain and the two workspaces set up in the config file (always there when `wezterm` starts)

**Reply by wez** [reply, marked as answer]
Can you file an issue for this? multiplexing is supposed to auto-reconnect. It's possible this got overlooked when considering ssh domains vs. tls or unix domains.

**Reply by mkuratczyk** [reply]
Done: https://github.com/wez/wezterm/issues/3539
Thanks a lot!
